Refactor

SimonBartonPSU opened the pull request:
- Move the proceed button listener to [gameObjects.js](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/master/gameObjects.js)
- Also made the move function more modular
- Wormhole check and direction check are separate functions
- Minor fixes with getElementById.

SimonBartonPSU committed:
button listener, separated switch statement and wormhole check from move()

Added a listener that will call spaceship.move for the proceed button instead of with onmousedown call in HTML.

Separated direction check switch statement and wormhole check in move to different functions.

Fixed the last getElementById pieces in gameObjects.

SimonBartonPSU committed:
Removed comment

removed an unnecessary comment

SimonBartonPSU committed:
Listener syntax change

Switched my listener from () => to function() syntax

halyngoc commented on gameObjects.js:
```diff
@@ -76,13 +83,19 @@ function setData() {
+}
+
+//Listener for 'Proceed' button
+document.getElementById("proceed").onclick = function()
```
I might be wrong but this could be shortened to
```js
document.getElementById("proceed").onclick = spaceship.move;
```

SimonBartonPSU replied:
For some reason that messes up the energy and supply value decreasing within the move function. The coordinates still update, but other stuff seems to break with that syntax change.

halyngoc replied:
Huh, I wonder why.

halyngoc replied:
I have a feeling this might be a bigger bug. Wanna open an issue on this and leave for later?

SimonBartonPSU committed:
Small fix

Removed an outdated comment, changed one line in wormhole check for random behavior

SimonBartonPSU commented:
Closing this PR, because all of the changes are already in Liam's PR.
